sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2511,6 +2511,27 @@ object GpuOverrides {
       (a, conf, p, r) => new ComplexTypeMergingExprMeta[Concat](a, conf, p, r) {
         override def convertToGpu(child: Seq[Expression]): GpuExpression = GpuConcat(child)
       }),
+    expr[ConcatWs](
+      "Concatenates multiple input strings or array of strings into a single " +
+        "string using a given separator",
+      ExprChecks.projectNotLambda(TypeSig.STRING, TypeSig.STRING,
+        repeatingParamCheck = Some(RepeatingParamCheck("input",
+          (TypeSig.STRING + TypeSig.ARRAY).nested(TypeSig.STRING),
+          (TypeSig.STRING + TypeSig.ARRAY).nested(TypeSig.STRING)))),
+      (a, conf, p, r) => new ExprMeta[ConcatWs](a, conf, p, r) {
+        override def tagExprForGpu(): Unit = {
+          if (a.children.size <= 1) {
+            // If only a separator specified and its a column, Spark returns an empty 
+            // string for all entries unless they are null, then it returns null.
+            // This seems like edge case so instead of handling on GPU just fallback.
+            willNotWorkOnGpu("Only specifying separator column not supported on GPU")
+          }
+        }
+        override final def convertToGpu(): GpuExpression = {
+          convertToGpu(childExprs.map(_.convertToGpu()))
+        }
+        def convertToGpu(child: Seq[Expression]): GpuExpression = GpuConcatWs(child)
```

Comment:
nit: Not really sure why this is separated into two functions.

```
override final def convertToGpu(): GpuExpression =
  GpuConcatWs(childExprs.map(_.convertToGpu()))
```

feels just as clear to me and is smaller.